src/videojs.ima.js

```diff
@@ -1092,7 +1092,22 @@
       clearInterval(updateTimeIntervalHandle);
       clearInterval(seekCheckIntervalHandle);
       clearInterval(resizeCheckIntervalHandle);
-      player.one('play', player.ima.setUpPlayerIntervals_);
+      if(player.el()) player.one('play', player.ima.setUpPlayerIntervals_);
+    };
+
+    var playerDisposedListener = function(){
+      contentEndedListeners, contentAndAdsEndedListeners = [], [];
+      contentComplete = true;
+      player.off('ended', localContentEndedListener);
+      if(updateTimeIntervalHandle){
+        clearInterval(updateTimeIntervalHandle);
+      }
+      if(seekCheckIntervalHandle){
+        clearInterval(seekCheckIntervalHandle);
+      }
+      if(adTrackingTimer){
+        clearInterval(adTrackingTimer);
+      }
```

Comment:
Also need to clear the resizeCheckIntervalHandle, call adsManager.destroy(), and maybe remove the adContainerDiv